charts/buffer/templates/ingress/backendconfig.yaml

```diff
@@ -1,22 +1,3 @@
-{{ if .Values.ingress.enabled -}}
-apiVersion: cloud.google.com/v1
-kind: BackendConfig
-metadata:
-  name: {{ .Values.name }}-ingress-backendconfig
-  labels:
-    {{- include "buffer.labels" . | nindent 4 }}
-spec:
-  timeoutSec: {{ .Values.ingress.timeoutSec }}
-  healthCheck:
-    checkIntervalSec: 10
-    timeoutSec: 5
-    healthyThreshold: 2
-    unhealthyThreshold: 2
-    type: HTTPS
-    port: 443
-    requestPath: /status
-  {{- if .Values.ingress.securityPolicy }}
-  securityPolicy:
-    name: {{ .Values.ingress.securityPolicy }}
-  {{- end }}
+{{- include "libchart.backendconfig" (list . "buffer.backendconfig" )  -}}
```

Comment:
what happens here if ingress is disabled and the library chart isn't included? What defines `libchart.backendconfig`?

Reply:
So the file now looks like this:
```
{{- define "buffer.ingress" -}}
{{- end }}
```
and in the values file `ingress.enabled: false`?

Reply:
Sorry, let me rephrase:
What happens if ingress is disabled? The ingress chart wouldn't exist as a dependency, so line 1 of
```
{{- include "libchart.backendconfig" (list . "buffer.backendconfig" ) -}}
{{- define "buffer.backendconfig" -}}
{{- end }}
```
would be referencing a "libchart.backendconfig" that doesn't exist. What happens then?

Reply:
Almost--I think we should put the `include` inside the `if`, not before it, because really what we want to say is "if ingress is disabled, don't even try to access the `libchart.backendconfig` it defines"

Reply:
The same issue applies [here](https://github.com/broadinstitute/terra-helm/pull/506#discussion_r716972208).

Reply:
Ah, maybe include works differently. I'd always used `template` which does support being inside a conditional https://github.com/broadinstitute/terra-helm/blob/85d1cf7b62ef30423896759c87ec6f5f66113dce/charts/rawls/templates/migration.yaml

Reply:
Yeah, they have different purposes. You can read more about it [here](https://helm.sh/docs/chart_template_guide/named_templates/#the-include-function). But in short: 

> Because template is an action, and not a function, there is no way to pass the output of a template call to other functions; the data is simply inserted inline.